bin/add_frontend_server.rb

```diff
@@ -1,15 +1,60 @@
 require 'aws-sdk'
 require_relative '../deployment'
+require 'net/ssh'
+require 'net/scp'
+require 'io/console'
+require 'json'
+require 'etc'
 
 MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
 
+def execute_ssh_on_channel(channel, command, exit_error_string)
+  channel.exec(command) do |ch|
+    ch.on_data do |_, data|
+      $stdout.print data
+    end
+
+    ch.on_extended_data do |_, _, data|
+      $stderr.print data
+    end
+
+    ch.on_request 'exit-status' do |_, data|
+      if data.read_long != 0
+        puts exit_error_string
+        exit(1)
+      end
+    end
+  end
+
+end
+
+options = {}
+
+OptionParser.new do |opts|
+  opts.on('-e', '--environment ENVIRONMENT', 'Environment to add frontend to') do |env|
+    options['environment'] = env
+  end
+
+  opts.on('-h', '--help', 'Print this') { puts options; exit }
+end.parse!
+
+raise OptionParser::MissingArgument, 'Environment is required' if options['environment'].nil?
+
+puts 'Type in your password for gateway.code.org'
```

Comment:
Design question:  it looks like this can be run anywhere and that it does an ssh to gateway to execute commands, which adds some complexity to the script.  

I think it would actually be fine to ask users to log into gateway and run the script from there; this is an easy step and most people on call will be connected anyway.  Will, perhaps you want to weigh in on this point as well.  OTOH I am fine with things as they are also.